### PR TITLE
Migrate source to ES6 import/export

### PR DESCRIPTION
**Pull request by RubenVerborgh**
This commit migrates the `src` folder to use `import`/`export` rather than `require`/`module.exports`. In addition to just using more modern semantics, it allows tree shaking by webpack and others.

While it uses `default` exports for _internal_ files, the external exports (`index.js`) do _not_ use `default`. Hence, external code using `require` keeps on working.

The only breaking change is that external code that was incorrectly using `import rdf from 'rdflib'` should use `import * as rdf from 'rdflib'`. Preferably, such code imports the exposed subcomponents of `rdflib`, such that tree shaking becomes possible.

### CODE REVIEW

**Comment by timbl** [comment]
Note that ES6 import/export if you use the 'default' version is NOT COMPATIBLE with nodejs/commonjs module systems.  Therefore if default export is used, this will be a breaking change to the API.    That would take a high level strategic discussion.   So I assume that you are not using export default.

**Comment by RubenVerborgh** [comment]
Note that the usage of defaults only happens in internal files; our exports
in index.js so not use default. Unless libraries directly depend on
internal files (which I hope not), this should be fine.

File: src/updates-via.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 /*
  * Updates-Via
  */
-const namedNode = require('./data-factory').namedNode
+import { namedNode } from './data-factory'
```
Comment (Vinnl) [inline src/updates-via.js]: There's a bug here, reported here: https://gitter.im/linkeddata/rdflib.js?at=5d5c1e77950718247717e691

`namedNode` is not a named export of `data-factory` - instead, it offers a default export of an object with a property `namedNode`.

Comment (RubenVerborgh) [reply src/updates-via.js]: True; the export is correct, the import is not. On it.

Comment (RubenVerborgh) [reply src/updates-via.js]: Fix in e7c5680122def0a2b1298fb5c7158203dd24c707.

**Pull request merged**